packages/generator/tests/acceptance/test_streaming_measurement_client.py

```diff
@@ -0,0 +1,95 @@
+import importlib.util
+import pathlib
+from types import ModuleType
+from typing import Generator
+
+import pytest
+from ni_measurement_plugin_sdk_service.measurement.service import MeasurementService
+
+from ni_measurement_plugin_sdk_generator.client import create_client
+from tests.utilities.discovery_service_process import DiscoveryServiceProcess
+from tests.utilities.measurements import streaming_data_measurement
+
+
+def test___measurement_plugin_client___measure___returns_output(
+    measurement_plugin_client_module: ModuleType,
+) -> None:
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    output_type = getattr(measurement_plugin_client_module, "Output")
+    expected_measure_output = output_type(
+        name="<Name>",
+        index=9,
+        data=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
+    )
+    measurement_plugin_client = test_measurement_client_type()
+
+    measure_output = measurement_plugin_client.measure()
+
+    assert measure_output == expected_measure_output
+
+
+def test___measurement_plugin_client___stream_measure___returns_output(
+    measurement_plugin_client_module: ModuleType,
+) -> None:
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    output_type = getattr(measurement_plugin_client_module, "Output")
+    measurement_plugin_client = test_measurement_client_type()
+
+    measure_output = measurement_plugin_client.stream_measure()
+
+    for count, output in enumerate(measure_output, start=0):
+        expected_measure_output = output_type(
+            name="<Name>",
+            index=count,
+            data=[i for i in range(count + 1)],
+        )
+        assert output == expected_measure_output
```

Comment:
Asserting in a loop loses information: which iteration failed?

Instead, build a list of expected outputs and a list of actual outputs, and compare them.